plugins/org.jkiss.dbeaver.ext.mysql/src/org/jkiss/dbeaver/ext/mysql/model/MySQLDialect.java

```diff
@@ -206,13 +206,24 @@ public boolean mustBeQuoted(String str, boolean forceCaseSensitive) {
     @Override
     protected String quoteIdentifier(@NotNull String str, @NotNull String[][] quoteStrings) {
         // Escape with first (default) quote string
-        return quoteStrings[0][0] + escapeString(str) + quoteStrings[0][1];
+        return quoteStrings[0][0] + escapeString(str, quoteStrings[0]) + quoteStrings[0][1];
     }
-    
+
     @NotNull
     @Override
     public String escapeString(String string) {
-        return string.replace("'", "''").replace("`", "``").replaceAll("\\\\(?![_%?])", "\\\\\\\\");
+        return escapeString(string, null);
+    }
+
+    @NotNull
+    protected String escapeString(@NotNull String string, @Nullable String[] quotes) {
```

Comment:
You can at least create a dialect test if you do not know how to make a datatransfer test. We already have PostgreDialectTest and OracleDialectTest. Please look at them.

Reply:
Done